LibSerialize.lua

```diff
@@ -215,24 +236,17 @@ The following serialization options are supported:
     table encountered during serialization. The function must return true for
     the pair to be serialized. It may be called multiple times on a table for
     the same key/value pair. See notes on reeentrancy and table modification.
-* `asyncMode`: `boolean` (default false)
-  * `true`: the serialize function will return a coroutine handler to process
-    the serialization. If true, the following additional options are considered:
-    * `yieldOnObjectCount`: `number` How large to allow the buffer before yielding
-    * `yieldOnElapsedTime`: `number` Max duration between yields
-    * `timeFn`: `function` To return the time in `number` format relevant to the
-      environment.
-  * `false`: the serialize function will return the serialized string directly.
-
-The following deserialization options are supported:
-* `asyncMode`: `boolean` (default false)
-  * `true`: the serialize function will return a coroutine handler to process
-    the serialization. If true, the following additional options are considered:
-    * `yieldOnObjectCount`: `number` How large to allow the buffer before yielding
-    * `yieldOnElapsedTime`: `number` Max duration between yields
-    * `timeFn`: `function` To return the time in `number` format relevant to the
-      environment.
-  * `false`: the serialize function will return the serialized string directly.
+When using `SerializeAsyncEx()`, these additional options are supported:
```

Comment:
Per my comments in the LibDeflate PR, I think these can be collapsed into a single option, where you supply either:

`yieldCheckFn: function(objectCount: number) => boolean` - a yield will occur if this function returns true, or
`yieldOnObjectCount: number` - a yield will occur after this number of objects

And then if neither are provided, yield after <hardcoded number> objects. With this approach, the library code just calls `yieldCheckFn` and does bookkeeping on the count of objects since yield, and the caller (or default args) handles the specifics. A time-based yield is then up to the caller to implement, which is easy (they just need to reset their own "elapsed" when returning true).

Reply:
The `yieldCheckFn` works well, makes `yieldOnObjectCount` redundant even since it's now all self contained.